posthog/temporal/usage_reports/run_usage_reports.py

```diff
@@ -0,0 +1,197 @@
+        capture_message(f"Usage reports are disabled for {inputs.at}")
+        return QueryUsageReportsResult(
+            org_reports={},
+            period=(datetime.now(), datetime.now()),  # Empty period
+        )
+
+    at_date = parser.parse(inputs.at) if inputs.at else None
+    period = get_previous_day(at=at_date)
+    period_start, period_end = period
+
+    org_reports = _get_all_org_reports(period_start, period_end)
+
+    return QueryUsageReportsResult(
+        org_reports=org_reports,
+        period=period,
+    )
+
+
+@activity.defn(name="send-usage-reports")
+async def send_usage_reports(
+    inputs: SendUsageReportsInputs,
+) -> None:
+    instance_metadata = get_instance_metadata(inputs.period)
+
+    at_date = parser.parse(inputs.at) if inputs.at else None
+    period_start, period_end = inputs.period
+
+    producer = None
+    try:
+        if settings.EE_AVAILABLE:
+            from ee.sqs.SQSProducer import get_sqs_producer
+
+            producer = get_sqs_producer("usage_reports")
+    except Exception:
+        pass
+
+    pha_client = get_ph_client(sync_mode=True)
+
+    total_orgs = len(inputs.org_reports)
+    total_orgs_sent = 0
+
+    pha_client.capture(
+        "internal_billing_events",
+        "organization usage report starting",
+        {
+            "total_orgs": total_orgs,
+            "region": get_instance_region(),
+        },
+        groups={"instance": settings.SITE_URL},
+    )
+
+    for org_report in inputs.org_reports.values():
+        try:
+            organization_id = org_report.organization_id
+
+            full_report = _get_full_org_usage_report(org_report, instance_metadata)
+            full_report_dict = _get_full_org_usage_report_as_dict(full_report)
+
+            # First capture the events to PostHog
+            if not inputs.skip_capture_event:
+                try:
+                    at_date_str = at_date.isoformat() if at_date else None
+                    capture_report(
+                        pha_client=pha_client,
+                        organization_id=organization_id,
+                        full_report_dict=full_report_dict,
+                        at_date=at_date_str,
+                    )
+                except Exception as err:
+                    logger.exception(f"Error capturing report for organization {organization_id}: {err}")
+
+            # Then send the reports to billing through SQS (only if the producer is available)
+            if has_non_zero_usage(full_report) and producer:
+                try:
+                    _queue_report(producer, organization_id, full_report_dict)
+                    total_orgs_sent += 1
+                except Exception as err:
+                    logger.exception(f"Error queueing report for organization {organization_id}: {err}")
+        except Exception as loop_err:
+            logger.exception(f"Error processing organization report: {loop_err}")
+
+    pha_client.capture(
```

Comment:
Do you want to force a flush of the client? Unsure of how the client is set up and whether it sends all the events immediately or not